### PR TITLE
Fixes #3676 Course Display Changes

### LINKED ISSUE #3676: Course Display Changes
## Motivation
With the additional data available in #3568 , it would be useful if the new instructor links are displayed (presently they are not). In the review of #3570 it was suggested that two changes should take place:

- the course catalog list should become clickable cards, as currently there is no way to normally reach the full course node screen
- instructor links should be displayed on the full node
- metadata about the course should potentially be displayed alongside the description in a card on the full node

### Is your feature request related to a problem? Please describe.
With the addition of instructor links, there is currently no ready way to see them.

### Describe the solution you'd like

- Clickable cards in course catalog
- Course metadata displayed in a card beside the description

### Describe alternatives you've considered
Not displaying the new fields.

### PR DESCRIPTION
**Pull request by tadean**
This PR provides the display changes that were formerly in #3570 

**Note:** this PR is based upon #3570 and that PR should be merged prior to this one

## Description

- This PR adds a side container for metadata beside the course description to the full node
- This PR makes the course catalog list clickable cards, so that there is now a path to reach the full node
- Instructor names added to full node display

## Related issues
#3676 
#3568 

## How to test
- Enable `az_demo` `az_http` and `az_course`
- log in and visit `/admin/config/az-quickstart/courses`
- Enter a single course that exists to import, such as `MATH 100`
- Click `Save and import courses`
- View the node that was created
- Verify that the instructor field appears, and that all links are text
- Select one of the netids that is an instructor. Do not choose `unassigned` if the course has a section with an unassigned instructor
- Select one of the demo content Person nodes and edit them. Give them a netID matching the netID you chose.
- Save the person node.
- Visit `/admin/config/az-quickstart/courses` again
- Click `Save and import courses`
- Verify that the node is updated.
- View the node and check if the netid you selected is now linked to the person.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [x] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My change requires release notes.

### CODE REVIEW

File: modules/custom/az_course/config/install/core.entity_view_display.node.az_course.az_card.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ third_party_settings:
       weight: 0
       format_type: html_element
       format_settings:
-        classes: 'card mb-4'
+        classes: 'card border-light card-clickable mb-4'
```
Comment (danahertzberg) [inline modules/custom/az_course/config/install/core.entity_view_display.node.az_course.az_card.yml:34]:
```suggestion
        classes: 'card card-clickable border hover mb-4'
```

File: modules/custom/az_course/config/install/core.entity_view_display.node.az_course.az_card.yml [hunk 2]
```diff
@@ -44,18 +44,32 @@ third_party_settings:
...
+        - group_text_muted
+      label: 'Card Link'
       parent_name: group_card
       region: content
       weight: 2
-      format_type: html_element
+      format_type: link
       format_settings:
-        classes: card-body
+        classes: 'card-body p-4'
```
Comment (danahertzberg) [inline modules/custom/az_course/config/install/core.entity_view_display.node.az_course.az_card.yml:54]:
```suggestion
        classes: 'card-body'
```

File: modules/custom/az_course/config/install/core.entity_view_display.node.az_course.az_card.yml [hunk 3]
```diff
@@ -70,7 +84,7 @@ third_party_settings:
       smart_title__tag: h5
       smart_title__classes:
         - card-title
-        - mt-0
+        - text-midnight
```
Comment (danahertzberg) [inline modules/custom/az_course/config/install/core.entity_view_display.node.az_course.az_card.yml:87]:
```suggestion
        - text-midnight
        - mt-1
        - hover-text-underline
```

**Review by danahertzberg — COMMENTED** [review]
Update card style

**Review by joeparsons — APPROVED** [review]

**Review by danahertzberg — APPROVED** [review]
Reviewed styles on the following pages
- https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-3677.probo.build/courses
- https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-3677.probo.build/course/math/100

Followed steps to ensure netID functionality still worked.

Everything looked good to me. 👍

**Comment by tadean** [comment]
Merged with changes from main after closing of #3568 to make diff easier to review.

**Pull request merged**